### PR TITLE
feat(vpn): add new data_source vpn gateways

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/vpn' TESTARGS='-run TestAccDatasourceGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccDatasourceGateway_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceGateway_basic
=== PAUSE TestAccDatasourceGateway_basic
=== CONT  TestAccDatasourceGateway_basic
--- PASS: TestAccDatasourceGateway_basic (318.44s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       318.517s
```

### CODE REVIEW

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/vpn' TESTARGS='-run TestAccDatasourceGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccDatasourceGateway_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceGateway_basic
=== PAUSE TestAccDatasourceGateway_basic
=== CONT  TestAccDatasourceGateway_basic
--- PASS: TestAccDatasourceGateway_basic (370.05s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       370.087s
```

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/vpn' TESTARGS='-run TestAccDatasourceGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccDatasourceGateway_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceGateway_basic
=== PAUSE TestAccDatasourceGateway_basic
=== CONT  TestAccDatasourceGateway_basic
--- PASS: TestAccDatasourceGateway_basic (269.18s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       269.218s
```

File: docs/data-sources/vpn_gateways.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+* `local_subnets` - The local subnets.
+
+* `connect_subnet` - The VPC network segment used by the VPN gateway.
+
+* `bgp_asn` - The ASN number of BGP
+
+* `flavor` - The flavor of the VPN gateway.
+
+* `availability_zone` - The availability zone IDs.
```
Comment (houpeng80) [inline docs/data-sources/vpn_gateways.md]:
```suggestion
* `availability_zones` - The availability zone IDs.

```
it is better to stay consistent with the resource, and fix the ohter places

**Pull request merged**